app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java

```diff
@@ -175,55 +170,53 @@ public static boolean almostEquals( TransmitterRawData e1, TransmitterRawData e2
     }
 
-    public static List<TransmitterRawData> ReadFromMongo(String dbury, int numberOfRecords)
-    {
-        Log.i(TAG,"Reading From " + dbury);
-    	List<TransmitterRawData> tmpList;
-    	// format is dburi/db/collection. We need to find the collection and strip it from the dburi.
-    	int indexOfSlash = dbury.lastIndexOf('/');
-    	if(indexOfSlash == -1) {
-    		// We can not find a collection name
-    		Log.e(TAG, "Error bad dburi. Did not find a collection name starting with / " + dbury);
-    		// in order for the user to understand that there is a problem, we return null
-    		return null;
-
-    	}
-    	String collection = dbury.substring(indexOfSlash + 1);
-    	dbury = dbury.substring(0, indexOfSlash);
-
-    	// Make sure that we have another /, since this is used in the constructor.
-    	indexOfSlash = dbury.lastIndexOf('/');
-    	if(indexOfSlash == -1) {
-    		// We can not find a collection name
-    		Log.e(TAG, "Error bad dburi. Did not find a collection name starting with / " + dbury);
-    		// in order for the user to understand that there is a problem, we return null
-    		return null;
-    	}
-
-    	MongoWrapper mt = new MongoWrapper(dbury, collection, "CaptureDateTime", "MachineNameNotUsed");
-    	return mt.ReadFromMongo(numberOfRecords);
+    public static List<TransmitterRawData> readFromMongo(String dbury, int numberOfRecords) {
+        Log.i(TAG, "Reading From " + dbury);
+        List<TransmitterRawData> tmpList;
+        // format is dburi/db/collection. We need to find the collection and strip it from the dburi.
+        int indexOfSlash = dbury.lastIndexOf('/');
+        if (indexOfSlash == -1) {
+            // We can not find a collection name
+            Log.e(TAG, "Error bad dburi. Did not find a collection name starting with / " + dbury);
+            // in order for the user to understand that there is a problem, we return null
+            return null;
+
+        }
+        String collection = dbury.substring(indexOfSlash + 1);
+        dbury = dbury.substring(0, indexOfSlash);
+
+        // Make sure that we have another /, since this is used in the constructor.
+        indexOfSlash = dbury.lastIndexOf('/');
+        if (indexOfSlash == -1) {
+            // We can not find a collection name
+            Log.e(TAG, "Error bad dburi. Did not find a collection name starting with / " + dbury);
+            // in order for the user to understand that there is a problem, we return null
+            return null;
+        }
+
+        MongoWrapper mt = new MongoWrapper(dbury, collection, "CaptureDateTime", "MachineNameNotUsed");
+        return mt.ReadFromMongo(numberOfRecords);
     }
 
     // format of string is ip1:port1,ip2:port2;
-    public static TransmitterRawData[] Read(String hostsNames, int numberOfRecords)
-    {
-        String []hosts = hostsNames.split(",");
-        if(hosts.length == 0) {
+    public static TransmitterRawData[] read(String hostsNames, int numberOfRecords) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method read() has an NPath complexity of 56](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l202)